src/sentry/static/sentry/app/components/gridEditable/grid.tsx

```diff
@@ -0,0 +1,283 @@
+import React from 'react';
+
+import {t} from 'app/locale';
+import {openModal} from 'app/actionCreators/modal';
+
+import Alert from 'app/components/alert';
+import EmptyStateWarning from 'app/components/emptyStateWarning';
+import InlineSvg from 'app/components/inlineSvg';
+import LoadingContainer from 'app/components/loading/loadingContainer';
+import {Panel, PanelBody} from 'app/components/panels';
+import ToolTip from 'app/components/tooltip';
+
+import {GridColumn, GridColumnOrder, GridColumnSortBy} from './gridTypes';
+import GridHeadCell from './gridHeadCell';
+import GridModalEditColumn from './gridModalEditColumn';
+import {
+  Grid,
+  GridRow,
+  GridHead,
+  GridBody,
+  GridBodyCell,
+  GridBodySpan,
+  GridEditGroup,
+  GridEditGroupButton,
+} from './styles';
+
+export type GridEditableProps<DataRow, ColumnKey extends keyof DataRow> = {
+  isEditable?: boolean;
+  isLoading?: boolean;
+  error?: React.ReactNode | null;
+
+  /**
+   * GridEditable (mostly) do not maintain any internal state and relies on the
+   * parent component to tell it how/what to render and will mutate the view
+   * based on this 3 main props.
+   *
+   * - `columnOrder` determines the columns to show, from left to right
+   * - `columnSortBy` is not used at the moment, however it might be better to
+   *   move sorting into Grid for performance
+   */
+  columnOrder: GridColumnOrder<ColumnKey>[];
+  columnSortBy: GridColumnSortBy<ColumnKey>[];
+  data: DataRow[];
+
+  /**
+   * GridEditable allows the parent component to determine how to display the
+   * data within it. Note that this is optional.
+   */
+  grid: {
+    renderHeaderCell?: {(column: GridColumnOrder<ColumnKey>): React.ReactNode};
+    renderBodyCell?: {
```

Comment:
What type is `renderBodyCell` supposed to be?

Reply:
The linter moved the type to line 52. Is that what you're looking for?

Reply:
Just curious about the syntax of the typing

Reply:
Fixed. Standardized to use `() => returnType` instead of `{ (): returnType }`.